scripts/testing/e2e-runner

```diff
@@ -0,0 +1,396 @@
+#!/bin/bash
+
```

Comment:
Added.

Reply:
Actually, now that I tested a few runs with a version updated with `set -eo pipefail`, this feels like a really bad idea. This script should survive as much as possible and try to collect any results it can get from a test run, instead of the opposite. I've tried to patch up all the failures I've been running into, but I'm seriously considering reverting this change...

Reply:
Okay, with the latest fixes, it now seems to successfully run till completion and is able to update the index.